fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACAuthorizingRealm.java

```diff
@@ -185,10 +186,17 @@ private Map<String, Collection<String>> getRolesForPath(final String path) {
 
     private Map<String, Collection<String>> getRolesForId(final FedoraId id) {
         Map<String, Collection<String>> roles = null;
-        final FedoraResource fedoraResource = getResourceOrParentFromPath(id);
-        if (fedoraResource != null) {
-            // check ACL for the request URI and get a mapping of agent => modes
-            roles = rolesProvider.getRoles(fedoraResource, transaction());
+
+        final var txId = FEDORA_ID_PREFIX + "/" + FCR_TX;
+        if (id.getResourceId().startsWith(txId)) {
+            final FedoraResource rootResource = getResourceOrParentFromPath(FedoraId.getRepositoryRootId());
```

Comment:
I'm considering changing this to `getResourceOrParentFromPath(id)`. In the event the id matches a tx, it will fallback to `info:fedora` since that will be first parent in its path. Probably not necessary just something I'm on the fence about.